frontend/src/scenes/surveys/SurveySQLHelper.tsx

```diff
@@ -0,0 +1,153 @@
+import { LemonButton, LemonDivider, LemonModal } from '@posthog/lemon-ui'
+import { useValues } from 'kea'
+import { router } from 'kea-router'
+import { CodeSnippet, Language } from 'lib/components/CodeSnippet'
+import { IconOpenInNew } from 'lib/lemon-ui/icons'
+import { surveyLogic } from 'scenes/surveys/surveyLogic'
+import { urls } from 'scenes/urls'
+
+import { NodeKind } from '~/queries/schema/schema-general'
+import { Survey, SurveyQuestion } from '~/types'
+
+import { createAnswerFilterHogQLExpression, getResponseFieldCondition } from './utils'
+
+interface SurveySQLHelperProps {
+    isOpen: boolean
+    onClose: () => void
+}
+
+export function SurveySQLHelper({ isOpen, onClose }: SurveySQLHelperProps): JSX.Element {
+    const { survey, answerFilters } = useValues(surveyLogic)
+
+    const filterConditions = createAnswerFilterHogQLExpression(answerFilters, survey as Survey)
+
+    const generateSingleQuestionQuery = (question: SurveyQuestion, index: number): string => {
+        return `SELECT
+    distinct_id,
+    ${getResponseFieldCondition(index, question.id)} AS "${question.question}",
+    timestamp
+FROM
+    events
+WHERE
+    event = 'survey sent'
+    AND properties.$survey_id = '${survey.id}'${filterConditions ? '\n' + filterConditions : ''}
```

Comment:
can we also use `coalesce` here instead of the `and` with the 2 keys

Reply:
not sure if I follow. `coalesce` is used to return the first non-null value, so why would we do that?

Reply:
my bad, i think i've mistaken the `$survey_id` with the question id here